core/src/main/java/bio/terra/pearl/core/service/search/terms/SearchValue.java

```diff
@@ -83,38 +83,57 @@ public static SearchValue ofNestedProperty(Object object, String field, SearchVa
         }
     }
 
+    private Object getValue() {
+        return switch (this.searchValueType) {
+            case STRING -> this.stringValue;
+            case NUMBER -> this.numberValue;
+            case INSTANT -> this.instantValue;
+            case DATE -> this.dateValue;
+            case BOOLEAN -> this.booleanValue;
+            case ARRAY -> this.arrayValue;
+            default -> null;
+        };
+    }
+
     public boolean equals(SearchValue right) {
         if (!this.searchValueType.equals(SearchValueType.ARRAY) && right.searchValueType.equals(SearchValueType.ARRAY)) {
             // flip the comparison if the right side is an array so that we hit the array comparison logic
             return right.equals(this);
         }
 
+        Object leftValue = this.getValue();
+        Object rightValue = right.getValue();
+
+        if (leftValue == null || rightValue == null) {
+            return leftValue == rightValue;
```

Comment:
`return leftValue == null && rightValue == null` has the same effect and might be easier to read